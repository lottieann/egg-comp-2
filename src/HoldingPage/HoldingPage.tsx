import * as React from 'react'

import styled from 'styled-components'

export interface HoldingPageProps {
    page: string
}

const Container = styled.div`
margin-top: 15vh;
display: flex;
height: 50vh;
justify-content: center;
align-items: center;
`

export const HoldingPage = (props: HoldingPageProps) => {
return (
   <Container>
       {props.page === "Votes" ? <p>Please return on Easter Saturday to cast your votes</p> :
       props.page === "Entries" ? <p>Please return on Easter Saturday to view the entries</p> : null}
   </Container>
)
}