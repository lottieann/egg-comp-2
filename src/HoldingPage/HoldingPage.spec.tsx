import * as React from 'react'

import 'jest-styled-components'

import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { HoldingPage, HoldingPageProps } from './HoldingPage'

const setup = (props: HoldingPageProps) => {
  return <HoldingPage {...props} />
}

describe('HoldingPage', () => {
  it('should render HoldingPage correctly on voting page', () => {
    const { container } = render(
      setup({
        page: 'Votes',
      }),
    )
    expect(container).toMatchSnapshot()
  })
  it('should render HoldingPage correctly on gallery page', () => {
    const { container } = render(
      setup({
        page: 'Entries',
      }),
    )
    expect(container).toMatchSnapshot()
  })
})
