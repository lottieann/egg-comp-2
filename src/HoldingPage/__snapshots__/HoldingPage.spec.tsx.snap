// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoldingPage should render HoldingPage correctly on gallery page 1`] = `
.c0 {
  margin-top: 15vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    <p>
      Please return on Easter Saturday to view the entries
    </p>
  </div>
</div>
`;

exports[`HoldingPage should render HoldingPage correctly on voting page 1`] = `
.c0 {
  margin-top: 15vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 50vh;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <div
    class="c0"
  >
    <p>
      Please return on Easter Saturday to cast your votes
    </p>
  </div>
</div>
`;
